### PR TITLE
OCM-4801 | feat: Link the enforce of billing account hcp for GA

### PR DESCRIPTION
**Pull request by lucasponce**
Discussed linking the enforcement of the request of the billing account flow for HCP once that GA is released.

Use the new technology preview API for coordination and flexibility.

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -1002,83 +1002,93 @@ func run(cmd *cobra.Command, _ []string) {
 		outputClusterAdminDetails(r, isClusterAdmin, clusterAdminPassword)
 	}
 
+	if isHostedCP && cmd.Flags().Changed("default-mp-labels") {
```
Comment (ciaranRoche) [inline cmd/create/cluster/cmd.go:1005]: Is this change releated to OCM-4801 or do you have a different card for this?

File: pkg/ocm/products.go [hunk 2]
```diff
@@ -23,12 +23,23 @@ func (c *Client) GetTechnologyPreview(id string) (*cmv1.ProductTechnologyPreview
...
+	return false, nil
+}
+
 func (c *Client) GetTechnologyPreviewMessage(id string, forTime time.Time) (string, error) {
 	techPreview, exists, err := c.GetTechnologyPreview(id)
 	if err != nil {
 		return "", err
 	}
-	// It assumes that the technology_preview endpoint will be available in CS before the ROSA CLI 1.2.30 is out
+	// It assumes that the technology_preview endpoint will be available in CS before the ROSA CLI HCP GA
```
Comment (andreadecorte) [inline pkg/ocm/products.go:42]: let's remove this line, endpoint is in production already

**Review by ciaranRoche — COMMENTED** [review]
Do we want to handle the case if its tech preview and someone passes --billing-account From what i can see here we will just silently ignore it :thinking: should we warn the user if they pass the param

File: cmd/create/cluster/cmd.go (hunk 1 above)
Comment (lucasponce) [reply cmd/create/cluster/cmd.go:1005]: It's a minor refactor, this logic was inside the billing block.

As it's unrelated I've extracted for readability.

(No new changes already)

**Comment by lucasponce** [comment]
> Do we want to handle the case if its tech preview and someone passes --billing-account From what i can see here we will just silently ignore it 🤔 should we warn the user if they pass the param

For this case there is a backend capability per org:

```
capability.organization.enforce_rosa_hcp_billing
```

So, that capability enforces if the billing account should be validated (at backend place).

File: cmd/create/cluster/cmd.go (hunk 1 above)
Comment (lucasponce) [reply cmd/create/cluster/cmd.go:1005]: https://github.com/openshift/rosa/pull/1599/files#diff-943c0ab57b82c7028652df55e1d49e4f4e4dc41fa094dc4a9b3f2c35887d44b5L1073

**Comment by andreadecorte** [comment]
lgtm

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**